include/opensyncmer.hpp

```diff
@@ -0,0 +1,545 @@
+    }
+
+private:
+    //!\brief The opensyncmer value.
+    value_type opensyncmer_value{};
+
+    //!\brief The offset relative to the beginning of the window where the opensyncmer value is found.
+    size_t opensyncmer_position_offset{};
+
+    //!\brief Iterator to the rightmost value of one kmer.
+    urng1_iterator_t urng1_iterator{};
+
+    //!\brief Iterator to the rightmost value of one kmer in the second range.
+    urng2_iterator_t urng2_iterator{};
+
+    //!brief Iterator to last element in range.
+    urng1_sentinel_t urng1_sentinel{};
+
+    //!\brief The number of values in one window.
+    size_t w_size{};
+
+    //!\brief Stored values per window. It is necessary to store them, because a shift can remove the current opensyncmer.
+    std::deque<value_type> window_values{};
+
+    //!\brief Increments iterator by 1.
+    void next_unique_opensyncmer()
+    {
+        while (!next_opensyncmer()) {}
+    }
+
+    //!\brief Returns new window value.
+    auto window_value() const
+    {
+        return *urng1_iterator;
+    }
+
+
+    //!\brief Advances the window to the next position.
+    void advance_window()
+    {
+        ++urng1_iterator;
+        ++urng2_iterator;
+    }
+
+    //!\brief Advances the first window to the next position.
+    void advance_first_window()
+    {
+        ++urng1_iterator;
+    }
+
+
+    //!\brief Calculates opensyncmers for the first window.
+    void window_first(const size_t K, const size_t S)
+    {
+	w_size = K - S + 1;
+
+	if (w_size == 0u)
+            return;
+
+        for (int i = 1u; i < K - 1 ; ++i)
+        {
+            window_values.push_back(window_value());
+            advance_first_window();
+        }
+        window_values.push_back(window_value());
+
+
+        auto smallest_s_it = std::ranges::min_element(window_values, std::less<value_type>{});
+	    opensyncmer_position_offset = std::distance(std::begin(window_values), smallest_s_it);
+
+	if (opensyncmer_position_offset == 0) {
+		auto opensyncmer_it = urng2_iterator;
+		opensyncmer_value = *opensyncmer_it;
+	}
+
+    }
+
+    /*!\brief Calculates the next opensyncmer value.
+     * \returns True, if new opensyncmer is found or end is reached. Otherwise returns false.
+     * \details
+     * For the following windows, we remove the first window value (is now not in window_values) and add the new
+     * value that results from the window shifting.
+     */
+    bool next_opensyncmer()
+    {
+    	advance_window();
+
+
+        if (urng1_iterator == urng1_sentinel)
+            return true;
+
+        value_type const new_value = window_value();
+
+        window_values.pop_front();
+        window_values.push_back(new_value);
+
+	if (opensyncmer_position_offset == 0)
+	{
+
+		auto smallest_s_it = std::ranges::min_element(window_values, std::less<value_type>{});
+
+		opensyncmer_position_offset = std::distance(std::begin(window_values), smallest_s_it);
+
+		if (opensyncmer_position_offset == 0) {
```

Comment:
The ; in the end seems wrong.
And seqan style is to put the {} on a  seperate line each

Reply:
I'm not sure I understand which one you mean. Could you please maybe highlight where exactly?